arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt

```diff
@@ -582,7 +582,7 @@ public sealed class Either<out A, out B> {
    * @param ifRight transform the [Either.Right] type [B] to [C].
    * @return the transformed value [C] by applying [ifLeft] or [ifRight] to [A] or [B] respectively.
    */
-  public inline fun <C> fold(ifLeft: (left: A) -> C, ifRight: (right: B) -> C): C {
+  public inline fun <C> fold(crossinline ifLeft: (left: A) -> C, crossinline ifRight: (right: B) -> C): C {
```

Comment:
Why crossinline? I think it's fine here to do a non-local return, no? Also, this stops `suspend` from transparently passing through